### DISCUSSION TITLE
Report | Week 02

### DISCUSSION THREAD

**Discussion by tech-n1c** [SE-TINF22B6/time2shine#15] [Show and tell]
## Overview
This week, we selected our Technology Stack and did our initial Setup, including:
- [Setup of a PostgreSQL Database in Docker](https://github.com/SE-TINF22B6/time2shine/issues/11)
- [Init of our Spring Boot Project, it will function as our Backend (as RESTful Web Service)](https://github.com/SE-TINF22B6/time2shine/issues/9)
- [Getting to know and testing OAuth2](https://github.com/SE-TINF22B6/time2shine/issues/10)
- [Init of a Java Gradle Project for the Frontend](https://github.com/SE-TINF22B6/time2shine/issues/14)

### Setup of Docker and PostgreSQL
We created a bwCloud instance, because we already know the tool due to prior lectures. It is only basic and the provided instances aren't particularly powerful, but they get the job done. Also, they are available for free and without any time limitation. After basic linux setup (e. g. installing sl), we installed Docker and created a Postgres Container. It will be used for storing user details, game results, leaderboards etc.

### Spring Boot Application
We initialized a new Spring Boot Project using the [Spring Initializr](https://start.spring.io) and agreed on the following:
- Java Version 17
- Gradle as our Build Tool
- Deployment in a BW-Cloud instance

After the initial setup of the project, we also set up the connection to our PostgreSQL database. In order to test this, we made up an account-entity (just as a test, because we'll use OAuth2 instead, as described below). A RestController was created to do basic CRUD-Operations and get comfortable with the Spring Environment. At that point, all these endpoints were directly accessible without any authorization, which is where OAuth2 comes in.

### OAuth2
We first did some research on the topic and found GitHub Authorization to be the most convenient way of authenticating users. Other Services like Google or Apple require complex setups and would make our setup more complex, meaning that we'd have to support all these different specifications. This is why we will stick to GitHub for the moment.

### Frontend
documentation to be done, research and testing is already in progress ( @sfrancisw ). Details will follow here until Thursday.

**Comment by sfrancisw** [comment]
### Frontend

We had a bit of a struggle to make a dummy java web application. We have cleared most of our problems and created a simple website that can be accessed via localhost. Next up, we'll start testing JSK (JavaServerFaces) to create an actual frontend. 

### Function

We've looked a bit at the game functions we have to implement and we'll start working on it once the frontend is ready.

**Comment by LeonmAndreesen** [comment]
Figma

As previous discussed we decided to do the frontend/interface with a programm called Figma. In the past week we had a superficial look at that tool. We discussed what kind of license is the one that we need to use. In that case it will be the educational license because it gives us more features. 
However, in order to implement some design we discuss and agree on what we want to implement in the game

**Comment by LPkkjHD** [comment]
Your Project looks very promising and ambitious. 

We are looking forwards to your implementation of OAuth especially. In addition we are curious on how you set up your project to be deployed on the BW-Cloud. And as per usual - don't let those grumpy SCRUM deniers rule your Backlog!

🧙🏿‍♂️Underwatch
